extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -361,6 +382,59 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     });
   }
 
+  /**
+   * Handles Fluid-related messages dispatched from SafeFrame.
+   * @param {!Event} event
+   * @private
+   */
+  receiveMessageForFluid_(event) {
+    const data = tryParseJson(getData(event));
+    if (event.origin != SAFEFRAME_ORIGIN || !data ||
+        data['s'] != 'creative_geometry_update') {
+      return;
+    }
+    user().assert(data['sentinel'] == this.sentinel,
+        `Expected sentinel value to match ${this.sentinel} but found` +
+        `${data['sentinel']}`);
+    // The first creative_geometry_update message will contain bad
+    // geometric data, as it will have been computed using the initial,
+    // incorrect, iframe style. We use this first message as a
+    // triggering point to restyle the iframe, which will in turn cause
+    // safeframe to send another creative_geometry_update message, this
+    // time containing the correct information.
+    if (this.iframe.style.width == '0px' &&
+        this.iframe.style.height == '0px') {
+      this.getVsync().mutate(() => {
+        setStyles(dev().assertElement(this.iframe), {
+          width: '100%',
+          height: '100%',
+          position: 'relative',
+        });
+        setStyles(this.element, {
+          width: computedStyle(this.win,
```

Comment:
`computedStyle` should be called in the measure phase.

Reply:
When the pub provided width is 'auto' we use getIntersectionElementLayoutBox to get the width.  Can we just do that here?

Reply:
That doesn't appear to be possible. getIntersectionElementLayoutBox returns 0 here, which makes sense, since the slot does in fact have width 0. What I'm doing with computedStyle is computing the width of the parent container of the ad slot, so that the ad slot can take up the maximum available space that it has.

Reply:
This still needs to be in a measure phase.

Reply:
Move setStyle to layoutMeasure.